src/util/addContext.js

```diff
@@ -8,7 +8,7 @@ export const addContext = (thought, context, rank) => {
   return Object.assign({}, thought, notNull({
     contexts: (thought.contexts || [])
       .filter(parent =>
-        !(equalArrays(parent.context, context) && parent.rank === rank)
```

Comment:
This breaks duplicate thoughts within the same context.

The user should still be able to manually add a duplicate thought to the same context:

- a
  - b
  - b

This PR currently adds the duplicate thought to the `contextIndex` but not the `thoughtIndex` because of the change on this line.

It is only when merging thoughts from a move that duplicates should be removed. You'll have to find a way to add this behavior without breaking the existing behavior of `addContext`.

Reply:
I was focused on merging, didn't think the user could add duplicated thought manually.
Right, after `addContext` and then remove duplicates.

Reply:
If merge b in the above case, then there's only one 'b' in 'a'?

Reply:
If a thought with children is merged into another thought with children, duplicate children should be discarded. If a user adds a duplicate thought manually, it should be properly added to `thoughtIndex` and `contextIndex` (as in `dev`).